packages/pirania/files/usr/bin/voucher

```diff
@@ -15,137 +12,93 @@ local context
 
 captive_portal = {}
 
-local function printJson (obj)
-    print(json.stringify(obj))
-end
+vouchera.init()
 
--- Show or change portal url
-captive_portal.url = function(context)
-    local url = ''
-    if (context[1] == 'show') then
-        url = uci_cursor:get("pirania", "base_config", "portal_url")
+captive_portal.activate = function(code, mac)
+    local res = vouchera.activate(code, mac)
+    if res then
+        print('Voucher activated!')
+        return true
     else
-        url = context[1] and context[1] or "http://thisnode.info/portal"
-        uci_cursor:set("pirania", "base_config", "portal_url", url)
-        uci_cursor:commit("pirania")
+        return nil, "Invalid voucher code '" .. code .. "' or invalid MAC '" .. mac .. "'"
     end
-    print(url)
 end
 
---[[
---Checks if a context(defined by a mac and a voucher code)
---is authorized to be used, and associates the mac to the
---voucher code if needed.
---]]
-captive_portal.auth_voucher = function(context)
-    local mac = context[1]
-    local voucher = context[2]
-
-    local db = dba.load(config.db)
-    local res = logic.auth_voucher(db, mac, voucher)
-    print ( res.limit[1], res.limit[2], res.limit[3], res.limit[4] )
-    if (res.success == true) then
-        print('Saving to DB')
-        print('...')
-            dba.save(config.db, db)
-        print('saved!')
+-- Checks if the mac of the given context is allowed to browse.
+captive_portal.is_mac_authorized = function(mac)
+    if vouchera.is_mac_authorized(mac) then
+        print('true')
+        return true
+    else
+        return nil, 'false'
```

Comment:
This is because how the processing of the functions from the main functions works ...it somewhat hacky.
I am pushing a clearer implementation, thanks.